src/uwtools/tests/drivers/test_chgres_cube.py

```diff
@@ -0,0 +1,213 @@
+            "mpiargs": ["--export=ALL", "--ntasks $SLURM_CPUS_ON_NODE"],
+            "mpicmd": "srun",
+        },
+        "namelist": {
+            "update_values": {
+                "config": {
+                    "atm_core_files_input_grid": "/path/to/file",
+                    "atm_files_input_grid": "/path/to/file",
+                    "atm_tracer_files_input_grid": "/path/to/file",
+                    "atm_weight_file": "/path/to/file",
+                    "convert_atm": True,
+                    "data_dir_input_grid": "/path/to/file",
+                    "external_model": "GFS",
+                    "fix_dir_target_grid": "/path/to/dir",
+                    "geogrid_file_input_grid": "/path/to/file",
+                    "grib2_file_input_grid": "/path/to/file",
+                    "mosaic_file_input_grid": "/path/to/file",
+                    "mosaic_file_target_grid": "/path/to/file",
+                }
+            }
+        },
+        "run_dir": "/path/to/dir",
+    },
+    "platform": {
+        "account": "me",
+        "scheduler": "slurm",
+    },
+}
+
+
+# Fixtures
+
+
+@fixture
+def cycle():
+    return dt.datetime(2024, 2, 1, 18)  # Fixtures
+
+
+@fixture
+def config_file(tmp_path):
+    path = tmp_path / "config.yaml"
+    config["chgres_cube"]["run_dir"] = tmp_path.as_posix()
+    with open(path, "w", encoding="utf-8") as f:
+        yaml.dump(config, f)
+    return path
+
+
+@fixture
+def driverobj(config_file, cycle):
+    return chgres_cube.ChgresCube(config_file=config_file, cycle=cycle, batch=True)
+
+
+# Driver tests
+
+
+def test_ChgresCube(driverobj):
+    assert isinstance(driverobj, chgres_cube.ChgresCube)
+
+
+def test_ChgresCube_dry_run(config_file, cycle):
+    with patch.object(chgres_cube, "dryrun") as dryrun:
+        driverobj = chgres_cube.ChgresCube(
+            config_file=config_file, cycle=cycle, batch=True, dry_run=True
+        )
+    assert driverobj._dry_run is True
+    dryrun.assert_called_once_with()
+
+
+def test_ChgresCube_namelist_file(driverobj):
+    @external
+    def ready(x):
+        yield x
+        yield asset(x, lambda: True)
+
+    src = str(driverobj._rundir / "input.nml.in")
+    dst = driverobj._rundir / "fort.41"
+    assert not dst.is_file()
+    with patch.object(chgres_cube, "file", new=ready):
+        driverobj._driver_config["namelist"]["update_values"]["config"] = {
+            "data_dir_input_grid": src,
+            "atm_files_input_grid": src,
+            "grib2_file_input_grid": src,
+            "sfc_files_input_grid": src,
+            "mosaic_file_target_grid": src,
+            "varmap_file": src,
+            "vcoord_file_target_grid": src,
+        }
+        driverobj.namelist_file()
+    assert dst.is_file()
+    assert isinstance(f90nml.read(dst), f90nml.Namelist)
+
+
+def test_ChgresCube_provisioned_run_directory(driverobj):
+    with patch.multiple(
+        driverobj,
+        namelist_file=D,
+        runscript=D,
```

Comment:
nitpick: I think if we remove the trailing comma on this line, `make format` will turn it into a single line again. It doesn't look long enough to need wrapping. `black` will fix it if that comma goes away.